--- a/usr.bin/env/env.c	2026-02-13 22:58:31.628861856 +0900
+++ b/usr.bin/env/env.c	2026-02-13 23:01:04.290460814 +0900
@@ -46,16 +46,12 @@
 
 #include <err.h>
 #include <errno.h>
-#include <login_cap.h>
-#include <pwd.h>
 #include <stdbool.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
 
-#include "envopts.h"
-
 extern char **environ;
 
 int	 env_verbosity;
@@ -72,26 +68,15 @@
 int
 main(int argc, char **argv)
 {
-	char *altpath, *altwd, **ep, *p, **parg, term;
+	char *altwd, **ep, *p, **parg, term;
 	char *cleanenv[1];
-	char *login_class, *login_name;
-	struct passwd *pw;
-	login_cap_t *lc;
-	bool login_as_user;
-	uid_t uid;
 	int ch, want_clear;
 	int rtrn;
 
-	altpath = NULL;
 	altwd = NULL;
-	login_class = NULL;
-	login_name = NULL;
-	pw = NULL;
-	lc = NULL;
-	login_as_user = false;
 	want_clear = 0;
 	term = '\n';
-	while ((ch = getopt(argc, argv, "-0C:iL:P:S:U:u:v")) != -1)
+	while ((ch = getopt(argc, argv, "-0C:iu:v")) != -1)
 		switch(ch) {
 		case '-':
 		case 'i':
@@ -103,22 +88,6 @@
 		case 'C':
 			altwd = optarg;
 			break;
-		case 'U':
-			login_as_user = true;
-			/* FALLTHROUGH */
-		case 'L':
-			login_name = optarg;
-			break;
-		case 'P':
-			altpath = optarg;
-			break;
-		case 'S':
-			/*
-			 * The -S option, for "split string on spaces, with
-			 * support for some simple substitutions"...
-			 */
-			split_spaces(optarg, &optind, &argc, &argv);
-			break;
 		case 'u':
 			if (env_verbosity)
 				fprintf(stderr, "#env unset:\t%s\n", optarg);
@@ -142,55 +111,6 @@
 		if (env_verbosity)
 			fprintf(stderr, "#env clearing environ\n");
 	}
-	if (login_name != NULL) {
-		login_class = strchr(login_name, '/');
-		if (login_class)
-			*login_class++ = '\0';
-		if (*login_name != '\0' && strcmp(login_name, "-") != 0) {
-			pw = getpwnam(login_name);
-			if (pw == NULL) {
-				char *endp = NULL;
-				errno = 0;
-				uid = strtoul(login_name, &endp, 10);
-				if (errno == 0 && *endp == '\0')
-					pw = getpwuid(uid);
-			}
-			if (pw == NULL)
-				errx(EXIT_FAILURE, "no such user: %s", login_name);
-		}
-		/*
-		 * Note that it is safe for pw to be null here; the libutil
-		 * code handles that, bypassing substitution of $ and using
-		 * the class "default" if no class name is given either.
-		 */
-		if (login_class != NULL) {
-			lc = login_getclass(login_class);
-			if (lc == NULL)
-				errx(EXIT_FAILURE, "no such login class: %s",
-				    login_class);
-		} else {
-			lc = login_getpwclass(pw);
-			if (lc == NULL)
-				errx(EXIT_FAILURE, "login_getpwclass failed");
-		}
-
-		/*
-		 * This is not done with setusercontext() because that will
-		 * try and use ~/.login_conf even when we don't want it to.
-		 */
-		setclassenvironment(lc, pw, 1);
-		setclassenvironment(lc, pw, 0);
-		if (login_as_user) {
-			login_close(lc);
-			if ((lc = login_getuserclass(pw)) != NULL) {
-				setclassenvironment(lc, pw, 1);
-				setclassenvironment(lc, pw, 0);
-			}
-		}
-		endpwent();
-		if (lc != NULL)
-			login_close(lc);
-	}
 	for (argv += optind; *argv && (p = strchr(*argv, '=')); ++argv) {
 		if (env_verbosity)
 			fprintf(stderr, "#env setenv:\t%s\n", *argv);
@@ -206,8 +126,6 @@
 		if (altwd && chdir(altwd) != 0)
 			err(EXIT_CANCELED, "cannot change directory to '%s'",
 			    altwd);
-		if (altpath)
-			search_paths(altpath, argv);
 		if (env_verbosity) {
 			fprintf(stderr, "#env executing:\t%s\n", *argv);
 			for (parg = argv, argc = 0; *parg; parg++, argc++)
@@ -222,8 +140,6 @@
 	} else {
 		if (altwd)
 			errx(EXIT_CANCELED, "must specify command with -C");
-		if (altpath)
-			errx(EXIT_CANCELED, "must specify command with -P");
 	}
 	for (ep = environ; *ep; ep++)
 		(void)printf("%s%c", *ep, term);
@@ -236,7 +152,7 @@
 usage(void)
 {
 	(void)fprintf(stderr,
-	    "usage: env [-0iv] [-C workdir] [-L|-U user[/class]] [-P utilpath] [-S string]\n"
-	    "           [-u name] [name=value ...] [utility [argument ...]]\n");
+	    "usage: env [-0iv] [-C workdir] [-u name]\n"
+	    "           [name=value ...] [utility [argument ...]]\n");
 	exit(1);
 }
