diff --git a/bin/cat/cat.c b/bin/cat/cat.c
--- a/bin/cat/cat.c
+++ b/bin/cat/cat.c
@@ -216,12 +216,14 @@
 			err(EXIT_FAILURE, "stdout");
 	}
 
+#ifndef BOOTSTRAP_CAT
 	init_casper(argc, argv);
 
 	caph_cache_catpages();
 
 	if (caph_enter_casper() != 0)
 		err(EXIT_FAILURE, "capsicum");
+#endif
 
 	if (bflag || eflag || nflag || sflag || tflag || vflag)
 		scanfiles(argv, 1);
@@ -259,11 +261,15 @@
 			fd = STDIN_FILENO;
 		} else {
 			filename = path;
+			#ifdef BOOTSTRAP_CAT
+			fd = open(path, O_RDONLY);
+			#else
 			fd = fileargs_open(fa, path);
 #ifndef NO_UDOM_SUPPORT
 			if (fd < 0 && errno == EOPNOTSUPP)
 				fd = udom_open(path, O_RDONLY);
 #endif
+			#endif
 		}
 		if (fd < 0) {
 			warn("%s", path);
