--- a/sbin/ping/ping.c	2026-03-28 03:32:03
+++ b/sbin/ping/ping.c	2026-03-28 03:32:14
@@ -76,8 +76,6 @@
 #include <netinet/ip_var.h>
 #include <arpa/inet.h>
 
-#include <libcasper.h>
-#include <casper/cap_dns.h>
 
 #ifdef IPSEC
 #include <netipsec/ipsec.h>
@@ -194,10 +192,8 @@
 static int interval = 1000;	/* interval between packets, ms */
 static int waittime = MAXWAIT;	/* timeout for each packet */
 
-static cap_channel_t *capdns;
 
 static void fill(char *, char *);
-static cap_channel_t *capdns_setup(void);
 static void pinger(void);
 static char *pr_addr(struct in_addr);
 static char *pr_ntime(n_time);
@@ -562,14 +558,14 @@
 	if (options & F_PINGFILLED) {
 		fill((char *)datap, payload);
 	}
-	capdns = capdns_setup();
+	
 	if (source) {
 		bzero((char *)&sock_in, sizeof(sock_in));
 		sock_in.sin_family = AF_INET;
 		if (inet_aton(source, &sock_in.sin_addr) != 0) {
 			shostname = source;
 		} else {
-			hp = cap_gethostbyname2(capdns, source, AF_INET);
+			hp = gethostbyname2(source, AF_INET);
 			if (!hp)
 				errx(EX_NOHOST, "cannot resolve %s: %s",
 				    source, hstrerror(h_errno));
@@ -597,7 +593,7 @@
 	if (inet_aton(target, &to->sin_addr) != 0) {
 		hostname = target;
 	} else {
-		hp = cap_gethostbyname2(capdns, target, AF_INET);
+		hp = gethostbyname2(target, AF_INET);
 		if (!hp)
 			errx(EX_NOHOST, "cannot resolve %s: %s",
 			    target, hstrerror(h_errno));
@@ -611,15 +607,6 @@
 	}
 
 	/* From now on we will use only reverse DNS lookups. */
-#ifdef WITH_CASPER
-	if (capdns != NULL) {
-		const char *types[1];
-
-		types[0] = "ADDR2NAME";
-		if (cap_dns_type_limit(capdns, types, 1) < 0)
-			err(1, "unable to limit access to system.dns service");
-	}
-#endif
 	if (connect(ssend, (struct sockaddr *)&whereto, sizeof(whereto)) != 0)
 		err(1, "connect");
 
@@ -1610,7 +1597,7 @@
 	if (!(options & F_HOSTNAME))
 		return inet_ntoa(ina);
 
-	hp = cap_gethostbyaddr(capdns, (char *)&ina, sizeof(ina), AF_INET);
+	hp = gethostbyaddr((char *)&ina, sizeof(ina), AF_INET);
 
 	if (hp == NULL)
 		return inet_ntoa(ina);
@@ -1667,30 +1654,4 @@
 	}
 }
 
-static cap_channel_t *
-capdns_setup(void)
-{
-	cap_channel_t *capcas, *capdnsloc;
-#ifdef WITH_CASPER
-	const char *types[2];
-	int families[1];
-#endif
-	capcas = cap_init();
-	if (capcas == NULL)
-		err(1, "unable to create casper process");
-	capdnsloc = cap_service_open(capcas, "system.dns");
-	/* Casper capability no longer needed. */
-	cap_close(capcas);
-	if (capdnsloc == NULL)
-		err(1, "unable to open system.dns service");
-#ifdef WITH_CASPER
-	types[0] = "NAME2ADDR";
-	types[1] = "ADDR2NAME";
-	if (cap_dns_type_limit(capdnsloc, types, 2) < 0)
-		err(1, "unable to limit access to system.dns service");
-	families[0] = AF_INET;
-	if (cap_dns_family_limit(capdnsloc, families, 1) < 0)
-		err(1, "unable to limit access to system.dns service");
-#endif
-	return (capdnsloc);
-}
+
